dev/fattest.simplicity/src/componenttest/topology/utils/tck/TCKUtilities.java

```diff
@@ -207,4 +220,542 @@ public static TCKJarInfo parseTCKDependencies(Type type, String[] dependencyOutp
+            mvnwFile.setExecutable(true, false);
+            if (TCKUtilities.isZos()) {
+                TCKUtilities.zosTagASCII(mvnwFile);
+            }
+        }
+
+        File targetFolder = new File(tckWorkingDir, ".mvn/wrapper");
+        Files.createDirectories(targetFolder.toPath());
+
+        File wrapperPropertiesFile = TCKUtilities.exportResource(targetFolder, "maven-wrapper.properties");
+        return wrapperPropertiesFile;
+    }
+
+    /**
+     * Download the maven wrapper jar to the correct directory within the TCK working directory
+     *
+     * @param  tckWorkingDir the TCK working directory
+     * @param  wrapperProps  the properties from the wrapper config file
+     * @param  authenticator the authenticator to use for the download
+     * @throws IOException
+     */
+    public static void downloadMvnWrapperJar(File tckWorkingDir, Properties wrapperProps, Authenticator authenticator) throws IOException {
+
+        String wrapperURL = wrapperProps.getProperty(MVN_WRAPPER_URL_KEY);
+        String wrapperHash = wrapperProps.getProperty(MVN_WRAPPER_SHA_KEY);
+        File targetFolder = new File(tckWorkingDir, ".mvn/wrapper");
+        Files.createDirectories(targetFolder.toPath());
+
+        TCKUtilities.downloadFile(wrapperURL, new File(targetFolder, "maven-wrapper.jar"), authenticator, wrapperHash);
+    }
+
+    public static void assertSHA256(File file, String expectedSha256) {
+        String actualSha256 = generateSHA256(file);
+        assertEquals(expectedSha256, actualSha256);
+    }
+
+    public static class ArtifactoryAuthenticator extends Authenticator {
+        PasswordAuthentication authentication;
+        String userName = getArtifactoryUser();
+        String password = getArtifactoryToken();
+
+        ArtifactoryAuthenticator() {
+            if (userName == null || password == null) {
+                authentication = null;
+            } else {
+                authentication = new PasswordAuthentication(userName, password.toCharArray());
+            }
+        }
+
+        @Override
+        protected PasswordAuthentication getPasswordAuthentication() {
+            return authentication;
+        }
+    }
+
+    /**
+     * Update the mvnw wrapper properties file to use the artifactory server defined in the
+     * environment. Specifically, this looks for the "distributionUrl" and "wrapperUrl" properties
+     * and replaces the public repository URL with the artifactory URL.
+     *
+     * @param  wrapperPropertiesFile the wrapper properties file to update
+     * @return                       the new set of properties
+     * @throws IOException
+     */
+    public static Properties updateWrapperPropertiesFile(File wrapperPropertiesFile) throws IOException {
+        Properties props = new Properties();
+        //get the artifactory server
+        String artifactoryServer = System.getProperty(FAT_TEST_PREFIX + ARTIFACTORY_SERVER_KEY);
```

Comment:
Pass in server and update doc to be clear this is only run when using artifactory